// redux-form-auto/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Schema", {
  enumerable: true,
  get: function () {
    return _schema.Schema;
  }
});
Object.defineProperty(exports, "Autoform", {
  enumerable: true,
  get: function () {
    return _Autoform.Autoform;
  }
});
Object.defineProperty(exports, "InputWrap", {
  enumerable: true,
  get: function () {
    return _InputWrap.InputWrap;
  }
});
Object.defineProperty(exports, "InputArrayWrap", {
  enumerable: true,
  get: function () {
    return _InputArrayWrap.InputArrayWrap;
  }
});
Object.defineProperty(exports, "translatable", {
  enumerable: true,
  get: function () {
    return _validation.translatable;
  }
});
Object.defineProperty(exports, "addValidator", {
  enumerable: true,
  get: function () {
    return _validators.addValidator;
  }
});
Object.defineProperty(exports, "addTranslations", {
  enumerable: true,
  get: function () {
    return _translate.addTranslations;
  }
});
Object.defineProperty(exports, "setTranslator", {
  enumerable: true,
  get: function () {
    return _translate.setTranslator;
  }
});
Object.defineProperty(exports, "tr", {
  enumerable: true,
  get: function () {
    return _translate.tr;
  }
});
Object.defineProperty(exports, "setLanguageByName", {
  enumerable: true,
  get: function () {
    return _translation_utils.setLanguageByName;
  }
});
Object.defineProperty(exports, "trModel", {
  enumerable: true,
  get: function () {
    return _translation_utils.trModel;
  }
});
Object.defineProperty(exports, "setSkin", {
  enumerable: true,
  get: function () {
    return _componentRender.setSkin;
  }
});
Object.defineProperty(exports, "addSkinType", {
  enumerable: true,
  get: function () {
    return _componentRender.addSkinType;
  }
});
Object.defineProperty(exports, "renderInput", {
  enumerable: true,
  get: function () {
    return _componentRender.renderInput;
  }
});
Object.defineProperty(exports, "renderInputs", {
  enumerable: true,
  get: function () {
    return _componentRender.renderInputs;
  }
});
Object.defineProperty(exports, "FieldPropsOverride", {
  enumerable: true,
  get: function () {
    return _componentRender.FieldPropsOverride;
  }
});

var _schema = require("./schema");

var _Autoform = require("./ui/Autoform");

var _InputWrap = require("./ui/wrappers/InputWrap");

var _InputArrayWrap = require("./ui/wrappers/InputArrayWrap");

var _validation = require("./validation");

var _validators = require("./validators");

var _translate = require("./translate");

var _translation_utils = require("./translation_utils");

var _componentRender = require("./ui/componentRender");
