// redux-form-auto/dist/ui/components/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = require("./Icon");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const renderContent = ({
  iconClass,
  text
}) => {
  if (iconClass) {
    return _react.default.createElement(_Icon.Icon, {
      className: iconClass,
      text: text
    });
  } else return text;
};

const buttonType = ({
  submit
}) => submit ? 'submit' : 'button';

const handleClick = ({
  disabled,
  onClick
}, e) => {
  if (!disabled && onClick) onClick(e);
};

const Button = ({
  text = '',
  onFocus,
  onClick,
  className,
  iconClass,
  submit,
  disabled,
  small,
  extraSmall
}) => {
  const buttonClass = className || 'btn-default';
  const classes = (0, _classnames.default)('btn', buttonClass, {
    disabled,
    'btn-sm': small,
    'btn-xs': extraSmall
  });
  return _react.default.createElement("button", {
    type: buttonType({
      submit
    }),
    className: classes,
    onClick: handleClick.bind(null, {
      disabled,
      onClick
    }),
    onFocus: onFocus
  }, renderContent({
    iconClass,
    text
  }));
};

exports.Button = Button;
Button.propTypes = {
  text: _propTypes.default.string,
  onClick: _propTypes.default.func,
  className: _propTypes.default.string,
  iconClass: _propTypes.default.string,
  submit: _propTypes.default.bool,
  disabled: _propTypes.default.bool
};
