// redux-form-auto/dist/ui/components/InputArrayPanel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputArrayPanel = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _Panel = require("./Panel");

var _Button = require("./Button");

var _renderLectures = require("./renderLectures");

var _translate = require("../../translate");

var _translation_utils = require("../../translation_utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const renderItemHeader = ({
  onRemove,
  fields,
  idx
}) => _react.default.createElement(_Button.Button, {
  "aria-hidden": "true",
  onClick: onRemove.bind(null, fields, idx),
  iconClass: "trash",
  className: "btn-danger",
  small: true
}, "\xD7");

const renderInputs = props => {
  const {
    fields,
    onRemove,
    children
  } = props;
  return fields.map((field, idx) => {
    const header = renderItemHeader({
      fields,
      onRemove,
      idx
    });
    return _react.default.createElement(_Panel.Panel, {
      key: idx,
      header: header
    }, props.instrumentChildren({
      field,
      idx,
      children
    }));
  });
};

const renderHeader = props => {
  const {
    schemaTypeName,
    fields,
    onAdd,
    newObject
  } = props;
  const boundAdd = onAdd.bind(null, fields, newObject);
  return _react.default.createElement("span", null, (0, _translation_utils.trModel)(schemaTypeName, fields.name) + ' ', _react.default.createElement(_Button.Button, {
    className: "btn-success",
    onClick: boundAdd,
    iconClass: "plus",
    text: (0, _translate.tr)('add'),
    small: true
  }));
};

const InputArrayPanel = props => {
  const {
    meta
  } = props;
  return _react.default.createElement(_Panel.Panel, {
    header: renderHeader(props)
  }, (0, _renderLectures.renderLectures)(meta), renderInputs(props));
};

exports.InputArrayPanel = InputArrayPanel;
InputArrayPanel.propTypes = {
  onAdd: _propTypes.PropTypes.func.isRequired,
  onRemove: _propTypes.PropTypes.func.isRequired,
  instrumentChildren: _propTypes.PropTypes.func.isRequired
};
