// redux-form-auto/dist/ui/components/mapRadioOptions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Radio = exports.mapRadioOptions = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reduxForm = require("redux-form");

var _translation_utils = require("../../translation_utils");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

const mapRadioOptions = ({
  name,
  schemaTypeName,
  fieldSchema: {
    options
  }
}) => options.map(op => {
  const label = (0, _translation_utils.trModel)(schemaTypeName, name, op);
  return _react.default.createElement("span", {
    key: op
  }, _react.default.createElement(_reduxForm.Field, {
    name: name,
    component: "input",
    type: "radio",
    value: op
  }), ` ${label} `);
});

exports.mapRadioOptions = mapRadioOptions;

class Radio extends _react.PureComponent {
  render() {
    return _react.default.createElement("div", {
      key: this.props.name
    }, this.props.children);
  }

}

exports.Radio = Radio;
