// redux-form-auto/dist/ui/components/Panel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Panel = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const renderHeader = header => {
  if (header) {
    return _react.default.createElement("div", {
      className: "panel-heading"
    }, header);
  } else return null;
};

const getClasses = ({
  panelType,
  className
}) => {
  const type = panelType || 'default';
  const buttonClass = className || '';
  return (0, _classnames.default)('panel', 'panel-' + type, buttonClass);
};

const Panel = ({
  header,
  table = null,
  className,
  panelType,
  onClick,
  children
}) => {
  const classes = getClasses({
    panelType,
    className
  });
  return _react.default.createElement("div", {
    className: classes,
    onClick: onClick
  }, renderHeader(header), _react.default.createElement("div", {
    className: "panel-body"
  }, children), table);
};

exports.Panel = Panel;
Panel.propTypes = {
  header: _propTypes.default.node,
  table: _propTypes.default.node,
  className: _propTypes.default.string,
  panelType: _propTypes.default.string,
  onClick: _propTypes.default.func
};
